src/braket/experimental/autoqasm/api.py

```diff
@@ -120,74 +114,26 @@ def _function_wrapper(
     if is_autograph_artifact(f):
         return f
 
-    # TODO: (laurecap) Update to accept a callback
-    wrapper_factory = _convert_program_wrapper(
-        user_config=user_config,
-        is_subroutine=is_subroutine,
-    )
-
-    return autograph_artifact(wrapper_factory(f))
-
-
-def _convert_program_wrapper(
-    user_config: aq_program.UserConfig,
-    is_subroutine: bool = False,
-) -> Callable:
-    """Generates a factory which does the conversion of a function into a callable
-    that returns a Program object containing the quantum program.
-
-    Args:
-        user_config (UserConfig): User-specified settings that influence program building.
-        is_subroutine (bool): If the function corresponds to a subroutine or main.
-
-    Returns:
-        Callable: a decorator that converts the given function into an equivalent
-        function that uses AutoQASM operations.
-    """
-    conversion_ctx = ag_ctx.NullCtx()
-
-    def _decorator(f: Callable) -> Callable[[Any], aq_program.Program]:
-        """aq.function decorator implementation."""
-
-        def _wrapper(*args, **kwargs) -> Union[aq_program.Program, Optional[oqpy.base.Var]]:
-            """Wrapper that calls the converted version of f."""
-            # This code is executed once the decorated function is called
-            options = converter.ConversionOptions(
-                recursive=False,
-                user_requested=True,
-                optional_features=_autograph_optional_features(),
-            )
-            return _convert_program(
-                f, conversion_ctx, options, user_config, args, kwargs, is_subroutine
-            )
-
-        if inspect.isfunction(f) or inspect.ismethod(f):
-            _wrapper = functools.update_wrapper(_wrapper, f)
-
-        decorated_wrapper = tf_decorator.make_decorator(f, _wrapper)
-        return autograph_artifact(decorated_wrapper)
+    if not wrapper_args:
+        wrapper_args = {}
 
-    return _decorator
-
-
-def _convert_gate_wrapper() -> Callable:
     def _decorator(f: Callable) -> Callable[[Any], None]:
-        """aq.gate decorator implementation."""
-
         def _wrapper(*args, **kwargs) -> Callable:
             """Wrapper that calls the converted version of f."""
             options = converter.ConversionOptions(
+                user_requested=True,
                 optional_features=_autograph_optional_features(),
             )
-            return _convert_gate(f, options, args, kwargs)
+            # Call the appropriate function converter
+            return callback(f, options, args, kwargs, **wrapper_args)
 
         if inspect.isfunction(f) or inspect.ismethod(f):
             _wrapper = functools.update_wrapper(_wrapper, f)
 
         decorated_wrapper = tf_decorator.make_decorator(f, _wrapper)
         return autograph_artifact(decorated_wrapper)
 
-    return _decorator
+    return autograph_artifact(_decorator(f))
```

Comment:
It looks like `_decorator` already calls `autograph_artifact` on its return value - can it be removed here?
```suggestion
    return _decorator(f)
```

Reply:
Oh, we actually don't need the `_decorator` function at all!